src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -225,6 +226,11 @@ typedef struct ucc_tl_mlx5_mcast_coll_comm {
     int                                     n_prep_reliable;
     int                                     n_mcast_reliable;
     int                                     wsize;
+    void                                   *group_setup_info;
```

Comment:
i think it is ok as it is right now. I do not want to add new structs

Reply:
why not add new struct? Instead of weird memcpy with offset you can just assing to struct fields, and instead of (sizeof(ucc_status_t) + mgid_s + mlid_s) just sizeof(struct_name)

Reply:
@Sergei-Lebedev what do you mean it's "weird"?

Reply:
What is the motivation to do 
```
memcpy((PTR_OFFSET(data, sizeof(ucc_status_t))),  &comm->event->param.ud.ah_attr.grh.dgid, mgid_s);
memcpy((PTR_OFFSET(data, sizeof(ucc_status_t) + mgid_s)),
                &comm->event->param.ud.ah_attr.dlid, mlid_s);
memcpy(data, &status, sizeof(ucc_status_t));
```
instead of 
```
data.gid = comm->event->param.ud.ah_attr.grh.dgid,;
data.lid = comm->event->param.ud.ah_attr.dlid;
data.status = status;
```

Reply:
do you think it is necessary right now when we have the code working fine? what's wrong with a memcpy? Assignment looks nicer? yes, but to what degree we wamt to hold on to a PR? Is it a performance issue with memcpy or your code style dictates that? I am afraid I won't spend more time on this PR. If you do not want to merge it, I will close it.

Reply:
> do you think it is necessary right now when we have the code working fine? 

Yes, I think it's necessary. We asked you 3 times why to use memcpy and haven't got an answer yet.

> what's wrong with a memcpy?

Somebody will have to maintain this code, it's not merge and forget. With memcpy it's not only less readable but also error prone.

> to what degree we wamt to hold on to a PR?

Till the moment the PR gets 2 approves and all tests passes.

>  Is it a performance issue with memcpy or your code style dictates that? 

It's not a performance issue, see the answer for "what's wrong with a memcpy?"

> I am afraid I won't spend more time on this PR. If you do not want to merge it, I will close it

It's ok.